omniledger/ChildTransactions.md

```diff
@@ -0,0 +1,120 @@
+Navigation: [DEDIS](https://github.com/dedis/doc/tree/master/README.md) ::
+[Cothority](https://github.com/dedis/cothority/tree/master/README.md) ::
+[Building Blocks](https://github.com/dedis/cothority/tree/master/BuildingBlocks.md) ::
+[OmniLedger](README.md) ::
+Child Transactions
+
+# Child Transactions
+
+This is an overview of how the transactions can be also accepted by child nodes
+and how the leader can fetch all new transactions from the children.
+
+## Transaction Queue and Block Generation
+
+This part of the document describes the technical details of the design and
+implementation of transaction queue and block generation for OmniLedger. The
+assumption is that the leader will not fail. We will implement view-change in
+the future, starting by eliminating stop-failure and then byzantine-failure.
+Further, we assume there exists a maximum block size of B bytes. Transaction
+Queue A transaction is similar to what is defined above, namely a key/kind/value
+triplet and a signature of the requester (client). However, for bookkeeping
+purposes, we add a private field: "status". A status can be one of three
+states: "New" | "Verified" | "Submitted". The transaction request message is
+the same as the Transaction struct above, e.g.
+
+```
+message TransactionRequest {
+  Transaction Transaction = 1;
+}
+```
+
+TransactionRequest can be sent to any conode. The client should send it to
+multiple conodes if it suspects that some of the conode might fail or are
+malicious. On the conode, it will store all transactions that it received, in a
+queue in memory, with the initial state being "New". We use a slice with a mutex
+as the implementation for the queue. If the total size of the queue exceeds B
+bytes (we may need to adjust this to support a large backlog), then the conode
+responds to the client with a failure message, otherwise a success message. The
+client should not see the success message as an indication that the transaction
+is included in the block, but that the transaction is received and may be
+accepted into a block. We do not attempt to check whether the transaction is
+valid at this point because the conode’s darc database might not be up-to-date,
+for example if it just came back online.  
+
+### Block Generation
+
+The poll method is inspired by the
+beta synchroniser, where the leader sends a message, e.g.
```

Comment:
What is beta synchronizer?

Reply:
No idea - this is copied from @kc1212 proposition to go further...